# HG changeset patch
# User Pantxo Diribarne <pantxo.diribarne@gmail.com>
# Date 1429568675 -7200
#      Tue Apr 21 00:24:35 2015 +0200
# Node ID de1377a638d7df0e6116febb1a149470d58ca6ca
# Parent  f1f6d5eab46f5a58c42fa58c5b66b6b8a9b2420e
Let annotation work with fltk again

* __init_fltk__.cc (fltk_graphics_toolkit::uimenu_set_fltk_label) silently omit menus who's parents are uicontextmenus.

diff --git a/libinterp/dldfcn/__init_fltk__.cc b/libinterp/dldfcn/__init_fltk__.cc
--- a/libinterp/dldfcn/__init_fltk__.cc
+++ b/libinterp/dldfcn/__init_fltk__.cc
@@ -2161,17 +2161,17 @@ public:
           dynamic_cast<uimenu::properties&> (uimenu_obj.get_properties ());
         std::string fltk_label = uimenup.get_label ();
         graphics_object go = gh_manager::get_object (uimenu_obj.get_parent ());
         if (go.isa ("uimenu"))
           fltk_label = dynamic_cast<const uimenu::properties&>
                        (go.get_properties ()).get_fltk_label ()
                        + "/"
                        + fltk_label;
-        else if (go.isa ("figure"))
+        else if (go.isa ("figure") || go.isa ("uicontextmenu"))
           ;
         else
           error ("unexpected parent object\n");
 
         uimenup.set_fltk_label (fltk_label);
       }
   }
 
